### DISCUSSION TITLE
How can I extract a code from a file?

### DISCUSSION THREAD

**Discussion by xoceunder** [NationalSecurityAgency/ghidra#6901] [Q&A]
How can I extract a code from a file .so

I want to be able to get a function

**Comment by ryanmkurtz** [comment]
Are you asking how to save the decompiler output for a function?

**Comment by xoceunder** [comment]
That's right, and if it's possible to see the real structure.

@ryanmkurtz

**Comment by xoceunder** [comment]
This Xui_Functions_updateCredentials

is file [https://file.io/cweOmW8X9YUP](url)

**Comment by ryanmkurtz** [comment]
To save the decompiled output of a function, you just click the icon that looks like a pencil and paper in the toolbar of the decompiler window.

**Reply by ryanmkurtz** [reply]
If you want the structures too, i think you need to export the entire program: `File -> Export Program -> Format: C/C++ -> Options -> Create Header File (.h)`

**Comment by xoceunder** [comment]
Could you help me show an illustration in a photo? I'm new to this.

**Reply by ryanmkurtz** [reply]
For just one function: 
![image](https://github.com/user-attachments/assets/79af12c1-cf9e-4915-89be-a48f3723c6cc)

For the entire program (including data types):
![image](https://github.com/user-attachments/assets/2e1da7b4-9608-4fbe-a24d-77dfd78a7a24)

**Comment by xoceunder** [comment]
I don't know where the last photo appears, what button do I have to press?

**Reply by ryanmkurtz** [reply]
`File -> Export Program`

**Reply by xoceunder** [reply]
> `File -> Export Program`

I can't find that

**Reply by ryanmkurtz** [reply]
It's in the Code Browser window...you have to import and open a program first.

**Comment by xoceunder** [comment]
As you can see the actual structure
